TorchSharp/TypeGeneration.tt

```diff
@@ -255,6 +257,34 @@ foreach (var type in new [] {
         {
             handle = TH<#=tname#>Tensor_newWithSize4d (size0, size1, size2, size3);
         }
+
+        /// <summary>
+        ///   Creates a 1-4D tensor of the specified size(s).
+        /// </summary>    
+        /// <param name="dims">Sizes for the dimensions.</param>
+        public <#=tname#>Tensor (params long[] dims)
+        {
+            switch (dims.Length)
+            {
+	            case 0:
+		            handle = TH<#=tname#>Tensor_new ();
+                    break;
+	            case 1: 
+		            handle = TH<#=tname#>Tensor_newWithSize1d (dims[0]);
+                    break;
+	            case 2: 
+		            handle = TH<#=tname#>Tensor_newWithSize2d (dims[0], dims[1]);
+		            break;
+	            case 3: 
+		            handle = TH<#=tname#>Tensor_newWithSize3d (dims[0], dims[1], dims[2]);
+		            break;
+	            case 4: 
+		            handle = TH<#=tname#>Tensor_newWithSize4d (dims[0], dims[1], dims[2], dims[3]);
+		            break;
+	            default:
+		            throw new ArgumentOutOfRangeException(nameof(dims), "Maximum number of dimensions for tensor creation is 4.");
```

Comment:
That's a good point. Is there a `reshape` function in TH? I am not able to find it.

Reply:
@interesaaat oh never mind then. we can commit it as it is, and then add `reshape` instead of an exception in some subsequent PR wehn we have such functionality

Reply:
Ok, marking as resolved. We will have to remember to fill an issue if we want to do the `reshape` thing.